examples/tabular/example_custom_feature_generator.py

```diff
@@ -0,0 +1,170 @@
+""" Example script for defining and using feature generators in AutoGluon Tabular """
```

Comment:
Briefly clarify what a feature generator is (eg. term 'data preprocessing' should probably appear somewhere in this description).

Reply:
Also may want to clarify which users might care about this example and why they might want to modify feature-generators (otherwise novice users may be confused into thinking they need to worry about this stuff).

Reply:
Maybe include link to sklearn fit_transform API to help users understand what is meant by "fitting" a feature generator

Reply:
added clarifications except for tke sklearn API comments. Will leave that for follow-up PR.